### DISCUSSION TITLE
IsPointerOverGameObject 가 동작하나요?

### DISCUSSION THREAD

**Discussion by Andwhy81kr** [naverz/zepeto-studio-kor#1167] [5. Q&A]
안녕하세요.

제페토에서 
EventSystem.current.IsPointerOverGameObject(Input.touches[0].fingerId)
이함수가 정상작동 하는지 알고 싶습니다.

에디터상에서 마우스를 이용한 IsPointerOverGameObject는 정상작동 하는게 확인이 되는데, 모바일에서 테스트 할땐 작동이 안되고 있는것 같습니다.

확인 부탁합니다.

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 저희 쪽에서 확인 결과 Unity 자체 이슈로 InputSystemUIInputModule 컴포넌트를 사용할 때, 
Touch 환경에서 EventSystem.IsPointerOverGameObject 를 통한 Touch 인식에 문제가 있음을 확인하였습니다.

제페토 프로젝트에서는 호환성을 위해 컨텐츠 씬에서 UI Component Input에 대해, 
InputSystemUIInputModule 컴포넌트(New Input System)로 일괄 처리하고 있습니다.
이 때, touch screen 환경에서
EventSystem.IsPointerOverGameObject 가 정상적이지 않은 이슈가 있습니다. (Unity 자체 이슈)

그래서 직접 raycast 통해 UI 에 대한 Input Event를 처리하는 방식으로 직접 핸들링이 가능합니다.
컨텐츠 단에서 직접 구현해서 사용하시는 방법을 추천 드립니다.

아래 예제 코드를 참고하여 주세요.
감사합니다.

```typescript
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { Input, Physics, RaycastHit, Vector2  } from "UnityEngine";
import {EventSystem, PointerEventData, RaycastResult } from 'UnityEngine.EventSystems';
import { List$1 } from 'System.Collections.Generic';

export default class IPointerTest extends ZepetoScriptBehaviour {
    
    public static IsPointerOverUIObject(position : UnityEngine.Vector2)
    {
        let vector2 : UnityEngine.Vector2 = position;
        let eventData : PointerEventData = new PointerEventData(EventSystem.current);
        eventData.position = position;

        let rayCastResults = new List$1<RaycastResult>();

        EventSystem.current.RaycastAll(eventData, rayCastResults);
        return rayCastResults.Count > 0;
    }
    
    Update(){
        if (Input.touches.length > 0
            && IPointerTest.IsPointerOverUIObject(Input.GetTouch(0).position)) {
            // UI touch Check
        }
    }
    
}
```
